scripts/test_screeps_rl_experiment_card.py

```
@@ -70,6 +71,198 @@ def test_generated_card_is_training_runner_valid(self) -> None:
+                        training_approach="policy_gradient",
+                        created_at="2026-05-17T00:25:00Z",
+                        registry_path=registry_path,
+                    )
+
+    def test_policy_gradient_registry_loader_errors_surface(self) -> None:
+        with tempfile.TemporaryDirectory() as temp_dir:
+            registry_path = Path(temp_dir) / "strategyRegistry.ts"
+            registry_path.write_text("export const STRATEGY_REGISTRY = [];\n", encoding="utf-8")
+
+            with mock.patch.object(runner, "load_strategy_registry", side_effect=RuntimeError("registry broke")):
+                with self.assertRaisesRegex(RuntimeError, "registry broke"):
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)